sdk/identity/src/token_credentials/options.rs

```diff
@@ -0,0 +1,76 @@
+use crate::env::Env;
+use azure_core::authority_hosts::AZURE_PUBLIC_CLOUD;
+use std::sync::Arc;
+use url::Url;
+
+const AZURE_AUTHORITY_HOST_ENV_KEY: &str = "AZURE_AUTHORITY_HOST";
+
+/// Provides options to configure how the Identity library makes authentication
+/// requests to Azure Active Directory.
+#[derive(Debug, Clone)]
+pub struct TokenCredentialOptions {
+    env: Env,
+    http_client: Arc<dyn azure_core::HttpClient>,
+    authority_host: Url,
+}
+
+/// The default token credential options.
+/// The authority host is taken from the `AZURE_AUTHORITY_HOST` environment variable if set.
+/// If not, the default authority host is `https://login.microsoftonline.com` for the Azure public cloud.
+impl Default for TokenCredentialOptions {
+    fn default() -> Self {
+        let env = Env::default();
+        let authority_host = env
+            .var(AZURE_AUTHORITY_HOST_ENV_KEY)
+            .map(|s| Url::parse(&s))
+            .unwrap_or_else(|_| Ok(AZURE_PUBLIC_CLOUD.to_owned()))
+            .unwrap_or_else(|_| AZURE_PUBLIC_CLOUD.to_owned());
```

Comment:
I am concerned this will swallow errors where the user intends to set the AZURE_AUTHORITY_HOST but it doesn't parse correctly.

Reply:
Okay, will have another look at this.

Reply:
I made it a `String` and `Url` parsing will fail in the `create`s.